crates/circuit/src/circuit_instruction.rs

```diff
@@ -30,16 +30,21 @@ use crate::operations::{
 };
 use crate::packed_instruction::PackedOperation;
 
-/// These are extra mutable attributes for a circuit instruction's state. In general we don't
-/// typically deal with this in rust space and the majority of the time they're not used in Python
-/// space either. To save memory these are put in a separate struct and are stored inside a
-/// `Box` on `CircuitInstruction` and `PackedInstruction`.
 #[derive(Debug, Clone)]
+struct ExtraAttrs {
+    label: Option<String>,
+    duration: Option<PyObject>,
+    unit: Option<String>,
+    condition: Option<PyObject>,
+}
```

Comment:
Two minor nits:

- imo for naming it'd be good to be consistent with either "attrs" or "attributes" between the two structs.  I know it's just internal to the module (hence just being a nit) and that we used `extra_attrs` as the attribute name, haha
- can you add a replacement doc comment explaining why the split exists, for the future?

Reply:
Done in bca33aa9443b0122a8411e057e40f63b7d5eb59f.